### DISCUSSION TITLE
Inquiry for @Home DSPL Simulation League Launch Sequence

### DISCUSSION THREAD

**Discussion by LeoYong95** [hsr-project/general#24] [General]
Dear @yosuke,

 In the competition, for both task 1 and task 2 all should have to launch in 1 launch file like in [here](https://github.com/devrt/robocup-at-home-2021-challenge/blob/main/robocup_challenge/launch/run.launch), or should there b 2 launch file one for task 1 and one for task 2?

based on the rule book competition flow:

> • Poster Session
> • Stage I
> - Clean Up (5 min)
> - Go and Get It (5 min)
> • Stage II
> - Clean Up (15 min)
> • Final

if both tasks share 1 launch file, I have the following questions.

1. Do the robot have to be at the room1 starting point b4 the start of Go and Get It? or the robot can move to the room2 entrance from wherever it is in Clean Up?
2. If the robot was not able to move to the room2 entrance in time, will the trail end?

if both task have separate launch file,  what is the sequence I should launch in the docker file?

Appreciate your time on answering my question...

**Comment by yosuke** [comment]
@LeoYong95 Sorry for the unclear announcement.

You are allowed use 20mins in the task.
If you can accomplish each tasks (task1, task2a, task2) in given time, you will get bonus points.
If your time exceeds 20mins in total, the task will be quit forcibly.
In the RCAP2021, all the tasks will be committed in serial.

So to answer your question,
> Do the robot have to be at the room1 starting point b4 the start of Go and Get It? or the robot can move to the room2 entrance from wherever it is in Clean Up?

The robot have to be at room1 when start, then start task1.
You can skip task1 to go task2, but in that case, no score or bonus points will be given for task1.

> If the robot was not able to move to the room2 entrance in time, will the trail end?

If the robot was able to move to the room2 within 20min time limit, there is still some chance to get a score.

**Reply by LeoYong95** [reply]
I see. Thank you so much for the info...

**Reply by LeoYong95** [reply]
Just to confirm:
the sequence of each task is:
1) task1
2) task2a
3) task2
total time for running this series of task is 20 min, there are no time constraint for each task as long as all task is completed within 20 min. Is this true?

**Reply by yosuke** [reply]
Yes, you have to keep the sequence (you cannot return to task1 after you enter room2).
For the time limits, it is also true. If you can finish task1 within 15min, and finish task2 in 5 min, you can get bonus points. While the overall time limit is 20min.
You can see whole the scoring algorithm from the following URL:
https://github.com/hsr-project/tmc_gazebo_task_evaluators/blob/master/src/wrs_score_counter.cpp

**Reply by LeoYong95** [reply]
I see.. Ok thanks for the info. ^^
